appservice/src/tree/AppSettingsTreeItem.ts

```diff
@@ -41,12 +47,15 @@ export class AppSettingsTreeItem extends AzExtParentTreeItem {
     public suppressMaskLabel: boolean = true;
     private _settings: StringDictionary | undefined;
     private readonly _settingsToHide: string[] | undefined;
+    public readonly contextValuesToAdd: string[];
 
-    constructor(parent: AzExtParentTreeItem, clientProvider: AppSettingsClientProvider, supportsSlots: boolean = true, settingsToHide?: string[]) {
+    constructor(parent: AzExtParentTreeItem, clientProvider: AppSettingsClientProvider, options?: AppSettingsTreeItemOptions) {
         super(parent);
         this.clientProvider = clientProvider;
-        this.supportsSlots = supportsSlots;
-        this._settingsToHide = settingsToHide;
+        this.supportsSlots = options?.supportsSlots ?? true;
+        this._settingsToHide = options?.settingsToHide;
+        this.contextValuesToAdd = options?.contextValuesToAdd ?? [];
+        this.contextValue = Array.from(new Set([AppSettingsTreeItem.contextValue, ...(options?.contextValuesToAdd ?? [])])).sort().join(';');
```

Comment:
Can this be a `get` with the `Set<string>` being constructed dynamically, like some of the other places?

Reply:
I think it's worth making this a util function, we're starting to use it a lot.

Reply:
Agreed, good idea.